[VTR Script] Add Run Number to Parse Script

amin1377 opened the pull request:
This PR introduces a run option to the parse script, allowing users to specify a specific run directory instead of always parsing the latest. If a run directory is provided, it will be parsed; otherwise, the script defaults to the latest run directory.

soheilshahrouz commented on vtr_flow/scripts/python_libs/vtr/util.py:
```diff
@@ -563,20 +574,26 @@ def get_latest_run_number(base_dir):
-    run_dir = Path(base_dir) / run_dir_name(run_number)
+    global global_run_dir_number
+    run_number = None
+    if global_run_dir_number is None:
```
I suggest you return `global_run_dir_number` if it is not `None` at the beginning of function.

```
if global_run_dir_number is not None:
    return global_run_dir_number
```

This way, the rest of function can be left untouched, and I think it would be easier to read.

soheilshahrouz commented on vtr_flow/scripts/python_libs/vtr/parse_vtr_task.py:
```diff
@@ -130,7 +131,7 @@ def vtr_command_argparser(prog=None):
     )
 
-    parser.add_argument("-run", default=None, type=str, help="")
+    parser.add_argument("-run", default=None, type=int, help="Run number to parse. If not provided, the latest run will be parsed.")
```
If you accept a string argument, the user can rename runxxx directories to more meaningful names and pass the run directory name directly.

amin1377 replied:
Good suggestion!

amin1377 commented:
@soheilshahrouz: I've addressed your comments. Please review the code again, as your feedback led to some significant changes. Let me know if you have any further suggestions.

soheilshahrouz commented on vtr_flow/scripts/python_libs/vtr/util.py:
```diff
@@ -549,34 +582,37 @@ def get_next_run_number(base_dir):
 
 
-def get_latest_run_number(base_dir):
+def get_latest_run_dir_name(base_dir):
```
The function name is confusing because the user provided directory name may not be the latest one.

soheilshahrouz commented on vtr_flow/scripts/python_libs/vtr/util.py:
```diff
@@ -549,34 +582,37 @@ def get_next_run_number(base_dir):
     """
-    run_number = 1
-    run_dir = Path(base_dir) / run_dir_name(run_number)
+    latest_run_dir_name = ""
```
The variable name can be confusing for the same reason.

soheilshahrouz commented on vtr_flow/scripts/python_libs/vtr/util.py:
```diff
@@ -225,6 +250,13 @@ def run_system_command(
+    """
+    RunDir.set_run_dir_name(current_run_dir_name)
+
+
```
Is this setter needed? 
`RunDir.set_run_dir_name()` is class method and does the same thing.

amin1377 replied:
Good point. I removed the function call and am calling the class method directly.

soheilshahrouz commented on vtr_flow/scripts/python_libs/vtr/util.py:
```diff
@@ -525,12 +557,13 @@ def get_latest_run_dir(base_dir):
     Returns the run directory with the highest run number in base_dir
     """
-    latest_run_number = get_latest_run_number(base_dir)
+    latest_run_dir_name = get_latest_run_dir_name(base_dir)
```
The function name, the comment and the variable name can be confusing becuase the user might specify another run directory.

amin1377 replied:
I changed the function name to `get_active_run_dir_name` and changed the comment to:

```
Returns the active run directory name. If the user has specified
a run directory name, it will be returned. Otherwise, the
highest run number of all run directories within base_dir
will be returned.
```
